src/speculators/train/eagle3/model_definitions.py

```diff
@@ -0,0 +1,119 @@
+# ruff: noqa: ERA001
+import copy
+from typing import NamedTuple
+
+import torch
+from transformers import Cache, LlamaConfig
+from transformers.models.llama.modeling_llama import (
+    LlamaDecoderLayer,
+    LlamaRMSNorm,
+    LlamaRotaryEmbedding,
+)
+from transformers.processing_utils import Unpack
+from transformers.utils.generic import TransformersKwargs
+
+
+class LlamaDecoderEagle3FirstLayer(LlamaDecoderLayer):
```

Comment:
Why do we call it FirstLayer?
This is the class for a single layer in the draft model, correct?